Engine/src/Engine/SceneBuilder/Scene.cpp

```diff
@@ -269,6 +266,19 @@ namespace Engine
 				glBufferData(GL_ARRAY_BUFFER, sizeof(vertices), vertices, GL_DYNAMIC_DRAW);
 			}
 
+			//load shader according to vertices component (given shader is not already loaded)
+			if (shaderNorms != nullptr && shaderNorms->queueStride(vertices.stride))
```

Comment:
this feels like it can be a function encapsulated in the ShaderNorms. Would help since we're trying to trim and decouple Scene

Reply:
You're correct that it can be encapsulated. I don't think it's very necessary though, the statement is very dependent on members of scene such as the vertices component and program Id as well as a growing list of indices length registers. The operation seems more dependent on scene than ShaderNorms.